src/components/HomeScreen.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import FlightList from "./FlightList";
+import useFlightApi from "../Apis";
+
+const HomeScreen = () => {
+  const history = useNavigate();
+
+  const [flightList, setFlightList] = useState([]);
+  const [filteredFlights, setFilteredFlights] = useState([]);
+  const [errorApi, setError] = useState(null); // State to store error information
+  const [showLoading, setShowLoading] = useState(true);
+
+  const { data, loading, error } = useFlightApi(
+    "https://flight-status-mock.core.travelopia.cloud/flights"
+  );
+
+  useEffect(() => {
+    if (!loading) {
+      setFilteredFlights([...data]);
+      setFlightList([...data])
+      setError(error);
+      setShowLoading(loading);
+    }
+  }, [data, loading, error]);
+
+  const fetchData = async () => {
+    try {
+      const response = await fetch(
+        "https://flight-status-mock.core.travelopia.cloud/flights"
+      );
+      if (!response.ok) {
+        throw new Error("Network response was not ok");
+      }
+      const data = await response.json();
+      setFilteredFlights([...data]);
+      setShowLoading(false);
+    } catch (error) {
+      setError(error.message);
+      setShowLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      // Fetch data every 30 seconds
+      fetchData();
+    }, 10000);
+
+    return () => clearInterval(intervalId); // Cleanup function to clear interval on component unmount
+  }, []);
+
+  const handleFlightClick = (id) => {
+    history(`/details/${id}`);
+  };
+
+  // Api call for list of flights
+
+  const handleSearch = (value) => {
+    if (value.trim() === "") {
+      setShowLoading(false);
+      return setFilteredFlights([...flightList]);
+    }
+    const filteredFlights = flightList.filter(
+      (flight) =>
+        (flight.airline.toLowerCase().startsWith(value.toLowerCase()) ||
+          flight.flightNumber.toLowerCase().startsWith(value.toLowerCase())) &&
+        flight
+    );
+    setFilteredFlights([...filteredFlights]);
+  };
+
+  const listReload = () => {
+    setFilteredFlights([...flightList]);
+    document.getElementById("searchInput").value = "";
+  };
+
+  return (
+    <div className="bg-container">
+      <div className="container-main">
+        <img
+          className="travelopia-logo"
+          src="https://www.drupal.org/files/travelopia_logo.png"
+          alt="Travelopia"
+        />
+        <input
+          className="inputSearch"
+          onChange={(e) => handleSearch(e.target.value)}
+          placeholder="Search for Airlines"
+          id="searchInput"
+        />
+        {errorApi ? ( // Display error message if there's an error
```

Comment:
this is too complicated. have three distinct blocks.

1. `error ? message : null`
2. `flightList.length ? errorForNoResponse : null`

it's not possible that error and flightList both would render.